Sources/MapboxSearch/PublicAPI/Common/Models/Country/Country.swift

```diff
@@ -12,4 +12,23 @@ public struct Country: Equatable {
 
         self.countryCode = countryCode.lowercased()
     }
+
+    init(code: ISO3166_1_alpha2) {
+        self.countryCode = code.rawValue.lowercased()
+    }
+
+    /// Detect the system region ISO3166\_1\_Alpha2 identifier and return an instance for it
+    static var `default`: Self? {
+        if #available(iOS 16, *) {
+            return Country(countryCode: Locale.current.region?.identifier ?? "")
+        } else {
+            let regionComponents = Locale.current.identifier.components(separatedBy: "_")
+            if regionComponents.count >= 2 {
+                let countryIdentifier = regionComponents[1]
```

Comment:
Hey @aokj4ck is `countryIdenitifier` only the last component in `regionComponents` or it is just the second item all the time? 

just smal nitpick from my side: is it possible to use `regionComponents?.last` or `regionComponents?.first` to leave all these checks on compiler shoulders? Instead to access item via index in array

Reply:
Great question, it **is** the second item all the time and there **may be** more items.

For example if you:
1. Add an iOS 15 simulator (oldest simulator available in Xcode 14 and that uses this API)
2. Change the Demo app > Edit Scheme > Run > Options > App Language and App Region
3. Set a breakpoint on this line
4. Build and run
5. Go to Discover tab and tap "Search in this area"

You will find that there are scenarios where the locale has more components such as Language = Chinese (Hong Kong) and Region = Hong Kong will return the locale `"en_HK_HK"`. There are various other examples and accessing the second items in the regionComponents is safest to cover all scenarios.

<img width="985" alt="Screenshot 2024-02-14 at 10 16 17" src="https://github.com/mapbox/mapbox-search-ios/assets/384288/6d0bd026-8a62-454a-a1dd-d5a7e08bb372">

<img width="960" alt="Screenshot 2024-02-14 at 10 18 44" src="https://github.com/mapbox/mapbox-search-ios/assets/384288/53fbf2cb-171f-4758-9c29-8a4c7e9e1489">